vector.go

```diff
@@ -424,12 +250,12 @@ func (vec *vector) initCString(t Type) {
 		}
 		return vec.getCString(rowIdx)
 	}
-	vec.setFn = func(vec *vector, rowIdx C.idx_t, val any) {
+	vec.setFn = func(vec *vector, rowIdx C.idx_t, val any) error {
 		if val == nil {
 			vec.setNull(rowIdx)
-			return
+			return nil
 		}
-		vec.setCString(rowIdx, val)
+		return setString(vec, rowIdx, val)
```

Comment:
I initially called this `initCString` and `setCString` to imply that this addresses both `VARCHAR` and `BLOB`.
It wasn't a good name, so maybe we can find a better one now. Otherwise, let's also rename `initCString` to `initString`.

Reply:
`setBytes` and `initBytes`? Since that is the underlying datastructure for both that seems sensible

Reply:
Yes, perfect 👍